### PR TITLE
Fix list view clipping issue

### PR DESCRIPTION
**Pull request by zilongshanren**
fixes issue #14357  #13759

### CODE REVIEW

**Comment by zilongshanren** [comment]
@dabingnn  @ricardoquesada 
please review it, thanks.

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
[ci rebuild]

File: tests/cpp-tests/Resources/ccs-res [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 636129882b612b5fe6d1d21be1eec136a2b1340b
+Subproject commit e82bd71212e53e016bf5f262f8e4bc1c50b54203
```
Comment (dabingnn) [inline tests/cpp-tests/Resources/ccs-res:1]: Did you commit it by accident?

Comment (zilongshanren) [inline tests/cpp-tests/Resources/ccs-res:1]: no, I added two new resources.

File: cocos/base/CCStencilBitsManager.hpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    StencilBitsManager();
+    StencilBitsManager(const StencilBitsManager& other);
+    StencilBitsManager& operator=(const StencilBitsManager& other);
+    GLint _stencilLayerMask;
+};
+
+NS_CC_END
+// end of base group
+/** @} */
+#endif /* StencilBitsManager_hpp */
```
Comment (dabingnn) [inline cocos/base/CCStencilBitsManager.hpp]: My suggestion is that we can refine it. The `StencilBitsManager` can act like a `stencil bits` pool, every one want `stencil bits` must try to get it, when it finished using it, the `stencil bits` must be returned to the `StencilBitsManager`.

Comment (zilongshanren) [inline cocos/base/CCStencilBitsManager.hpp]: @dabingnn 
Could you give a more concise API example?

**Comment by zilongshanren** [comment]
@dabingnn
Now I use `StencilStateManager` to handle various stencil buffer values and the stencil layer mask is a private static member of StencilStateManager class.

The code is simpler compared to the original implementation of `UILayout` and `ClippingNode` and no singletons are needed any more. :smile: 

please review it. thanks.

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by ricardoquesada** [comment]
Does this PR include the tests to reproduce issues #14357 #13759 ?
Is it backward compatible ?

**Comment by zilongshanren** [comment]
@ricardoquesada 
The tests are laid down in ListViewTest.cpp and it is definitely backward compatible.

**Comment by zilongshanren** [comment]
[ci rebuild]

File: tests/cpp-tests/Classes/UITest/CocoStudioGUITest/UIListViewTest/UIListViewTest.cpp [hunk 3]
```diff
@@ -123,6 +123,16 @@ bool UIListViewTest_Vertical::init()
             custom_item->addChild(custom_button);
             
             listView->addChild(custom_item);
+
```
Comment (zilongshanren) [inline tests/cpp-tests/Classes/UITest/CocoStudioGUITest/UIListViewTest/UIListViewTest.cpp:126]: @ricardoquesada 
I added the tests here.

**Comment by semgilo** [comment]
@zilongshanren  can i fix it base on  v3.9 
it base on which version ?

**Comment by zilongshanren** [comment]
based on v3.9

**Comment by zilongshanren** [comment]
@semgilo 
But this PR have a flaw, it doesn't free the memory of StencilStateManager.
Please pay attention to https://github.com/cocos2d/cocos2d-x/pull/14665

**Comment by semgilo** [comment]
@zilongshanren v3.10 or v3 version is fixed?3q

**Comment by zilongshanren** [comment]
@semgilo  v3.0 is fixed.

**Comment by semgilo** [comment]
@zilongshanren do u have QQ ? i had tried to fix base on v3.9  clipping is work but  position is wrong

**Comment by semgilo** [comment]
@zilongshanren i had fixed it, now i got a new isse, clippingnode isn't fadein or fadeout

**Pull request merged**